src/node/share_manager.h

```diff
@@ -359,9 +404,22 @@ namespace ccf
             }
           }
           OPENSSL_cleanse(decrypted_share.data(), decrypted_share.size());
+          if (full_share.has_value())
+          {
+            return false;
+          }
+
           return true;
         });
 
+      if (full_share.has_value())
+      {
+        // TBD (gsinha): In this variation of the LedgerSecretWrappingKey
+        // constructor do we set
+        //  shares and threshold member variables to any value?
+        return LedgerSecretWrappingKey(full_share.value());
```

Comment:
I don't believe we need to, but that's a good point and probably a reason to split the type for safety.

Reply:
Lets consider this in a separate PR if we do decide to split the type.